packages/react/src/events.ts

```diff
@@ -39,3 +40,14 @@ export function action<T = any>(name: T): ActionDataState<T> {
     name,
   }
 }
+
+export function withState<T extends Record<string, any>>(
+  state: { [K in keyof T]: StateDataState<T[K]> },
+  func: HandlerFunction<T>
+): ServerHandlerDataState {
+  return {
+    $: 'handler',
```

Comment:
hm, can you clarify the difference between $: "handler" and "event"?

Reply:
handler is just a special type that can be passed to `event`. Event takes either function or this object. When this object is present, it means you want to populate args of the `func` with `state`. Perhaps we could think of a better name, e.g. `func-with-state` or similar?

Reply:
hmm, maybe I was confused because this function returns a "ServerHandlerDataState" when it is not really a "DataState"

I thought a DataState is the thing sent to the client, but this is not ever sent to the client, it is only used internally to the server

Reply:
Agreed! Removed that type as well, and in the future, we might need a better name to differentiate across types that may loose certain properties while being serialised.